deeprankcore/features/surfacearea.py

```diff
@@ -1,17 +1,57 @@
 import logging
 import freesasa
-from deeprankcore.models.graph import Graph
-from deeprankcore.models.structure import Residue, Atom
-from deeprankcore.domain.features import nodefeats
+import numpy as np
+from typing import List
+from deeprankcore.utils.graph import Node, Graph
+from deeprankcore.molstruct.residue import Residue
+from deeprankcore.molstruct.atom import Atom
+from deeprankcore.domain import nodefeatures as Nfeat
 
 freesasa.setVerbosity(freesasa.nowarnings) # pylint: disable=c-extension-no-member
 logging.getLogger(__name__)
 
 
+def add_sasa_for_residues(structure: freesasa.Structure, # pylint: disable=c-extension-no-member
+                              result: freesasa.Result, # pylint: disable=c-extension-no-member
+                              nodes: List[Node]):
+
+    for node in nodes:
+
+        residue = node.id
+
+        selection = ("residue, (resi %s) and (chain %s)" % (residue.number_string, residue.chain.id),) # pylint: disable=consider-using-f-string
+
+        area = freesasa.selectArea(selection, structure, result)['residue'] # pylint: disable=c-extension-no-member
+        if np.isnan(area):
+            raise ValueError(f"freesasa returned {area} for {residue}")
+
+        node.features[Nfeat.SASA] = area
+
+
+def add_sasa_for_atoms(structure: freesasa.Structure, # pylint: disable=c-extension-no-member
+                           result: freesasa.Result, # pylint: disable=c-extension-no-member
+                           nodes: List[Node]):
+
+    for node in nodes:
+
+        atom = node.id
+
+        selection = ("atom, (name %s) and (resi %s) and (chain %s)" % \
+            (atom.name, atom.residue.number_string, atom.residue.chain.id),) # pylint: disable=consider-using-f-string
+
+        area = freesasa.selectArea(selection, structure, result)['atom'] # pylint: disable=c-extension-no-member
+        if np.isnan(area):
+            raise ValueError(f"freesasa returned {area} for {atom}")
+
+        node.features[Nfeat.SASA] = area
+
+
 def add_features(pdb_path: str, graph: Graph, *args, **kwargs): # pylint: disable=too-many-locals, unused-argument
 
-    "calculates the buried surface area (BSA): the area of the protein, that only gets exposed in monomeric state"
+    """calculates the Buried Surface Area (BSA) and the Solvent Accessible Surface Area (SASA):
+    BSA: the area of the protein, that only gets exposed in monomeric state"""
 
+    # BSA

```

Comment:
This section of code is quite long. Is there any possibility to make an `add_bsa` function?